e2e-test/cypress/integration/workflowitem_permissions_spec.js

```diff
@@ -1,82 +1,420 @@
+    cy.get("[data-test=permission-submit]").click();
+    actionTableIncludes("view permissions");
+  });
+
+  it("Granting view permissions doesn't additionally view the same permission", function() {
+    cy.grantProjectPermission(projectId, "project.viewDetails", testUser.id);
+    cy.grantProjectPermission(projectId, "project.viewSummary", testUser.id);
+    cy.grantSubprojectPermission(projectId, subprojectId, "subproject.viewDetails", testUser.id);
+    cy.grantSubprojectPermission(projectId, subprojectId, "subproject.viewSummary", testUser.id);
+
+    cy.get("[data-test=show-workflowitem-permissions]")
+      .first()
+      .click();
+    // Add permission
+    changePermissionInGui("workflowitem.view", testUser.id);
+    cy.get("[data-test=permission-submit]").click();
+    cy.get("[data-test=confirmation-dialog-cancel]").should("be.visible");
+    cy.get("[data-test=actions-table-body]").should("not.be.visible");
+
+    //Reset permissions
+    Cypress.Promise.all([
+      cy.revokeProjectPermission(projectId, "project.viewDetails", testUser.id),
+      cy.revokeProjectPermission(projectId, "project.viewSummary", testUser.id),
+      cy.revokeSubprojectPermission(projectId, subprojectId, "subproject.viewDetails", testUser.id),
+      cy.revokeSubprojectPermission(projectId, subprojectId, "subproject.viewSummary", testUser.id)
+    ]);
+  });
+
+  it("Executing additional actions grant permissions correctly", function() {
+    const listIntent = "workflowitem.intent.listPermissions";
+
+    cy.get("[data-test=show-workflowitem-permissions]")
+      .first()
+      .click();
+    // Add permission
+    changePermissionInGui("workflowitem.intent.revokePermission", testUser.id);
+    cy.get("[data-test=permission-submit]").click();
+    cy.get("[data-test=actions-table-body]")
+      .should("be.visible")
+      .children()
+      .should("have.length", 6);
+    cy.get("[data-test=confirmation-dialog-confirm]")
+      .click()
+      .should("be.not.disabled");
+
+    checkPermissionsEquality(
+      addViewPermissions(permissionsBeforeTesting, testUser.id),
+      projectId,
+      subprojectId,
+      workflowitemId
+    );
+
+    // Reset permissions
+    Cypress.Promise.all([
+      cy.revokeProjectPermission(projectId, "project.viewSummary", testUser.id),
+      cy.revokeProjectPermission(projectId, "project.viewDetails", testUser.id),
+      cy.revokeSubprojectPermission(projectId, subprojectId, "subproject.viewSummary", testUser.id),
+      cy.revokeSubprojectPermission(projectId, subprojectId, "subproject.viewDetails", testUser.id),
+      cy.revokeWorkflowitemPermission(projectId, subprojectId, workflowitemId, listIntent, testUser.id),
+      cy.revokeWorkflowitemPermission(projectId, subprojectId, workflowitemId, "workflowitem.view", testUser.id)
+    ]);
+  });
+
+  it("Granting assign/grant/revoke permissions addtionally views 'list permissions'-permissions", function() {
```

Comment:
views -> shows? Description not clear

Reply:
I changed the description as follows: "Granting assign/grant/revoke permissions additionally generates an action to grant 'list permissions'-permissions"

Reply:
ok